sdk/go/common/resource/plugin/analyzer_plugin.go

```diff
@@ -579,6 +579,25 @@ func convertURNs(urns []resource.URN) []string {
 	return result
 }
 
+func convertAlias(alias resource.Alias) *pulumirpc.Alias {
+	isUrn := alias.URN != ""
+	if isUrn {
+		return &pulumirpc.Alias{Alias: &pulumirpc.Alias_Urn{Urn: string(alias.URN)}}
+	}
+	return &pulumirpc.Alias{Alias: &pulumirpc.Alias_Spec_{Spec: &pulumirpc.Alias_Spec{Name: alias.Name}}}
+}
+
+func convertAliases(aliases []resource.Alias, aliasURNs []resource.URN) []*pulumirpc.Alias {
+	result := make([]*pulumirpc.Alias, len(aliases)+len(aliasURNs))
+	for idx, alias := range aliases {
+		result[idx] = convertAlias(alias)
+	}
+	for idx, aliasURN := range aliasURNs {
+		result[idx+len(aliases)] = convertAlias(resource.Alias{URN: aliasURN})
+	}
+	return result
```

Comment:
I'm not sure what `golang` convention is re: pre-allocating array sizes vs appending to variable-length arrays.  I tend to err towards statically-sized arrays when possible, but if this is too hacky looking I can revise.